MdeModulePkg/Library/MemoryProtectionHobLib/DxeMemoryProtectionHobLib.c

```diff
@@ -171,6 +171,95 @@ inactive but RaiseErrorIfProtectionFails is active. RaiseErrorIfProtectionFails
       ));
     gDxeMps.ImageProtectionPolicy.Fields.RaiseErrorIfProtectionFails = 0;
   }
+
+  //
+  // the heap guard system does not support non-EFI_PAGE_SIZE alignments
+  // architectures that require larger RUNTIME_PAGE_ALLOCATION_GRANULARITY
+  // cannot have EfiRuntimeServicesCode, EfiRuntimeServicesData, EfiReservedMemoryType,
+  // and EfiACPIMemoryNVS guarded. OSes do not map guard pages anyway, so this is a
+  // minimal loss. Not guarding prevents alignment mismatches
+  //
+  if (RUNTIME_PAGE_ALLOCATION_GRANULARITY != EFI_PAGE_SIZE) {
+    if (gDxeMps.HeapGuardPageType.Fields.EfiACPIMemoryNVS != 0) {

```

Comment:
Can you also do a check against gDxeMps.HeapGuardPolicy.Fields.UefiPageGuard (the global page guard toggle) before printing the page guard warnings?